azurelinuxagent/common/version.py

```diff
@@ -75,22 +75,26 @@ def get_checkpoint_platform():
 
 def get_distro():
     if 'FreeBSD' in platform.system():
-        release = re.sub('\-.*\Z', '', ustr(platform.release()))
+        release = re.sub(r'\-.*\Z', '', ustr(platform.release()))
         osinfo = ['freebsd', release, '', 'freebsd']
     elif 'OpenBSD' in platform.system():
-        release = re.sub('\-.*\Z', '', ustr(platform.release()))
+        release = re.sub(r'\-.*\Z', '', ustr(platform.release()))
         osinfo = ['openbsd', release, '', 'openbsd']
     elif 'Linux' in platform.system():
         osinfo = get_linux_distribution(0, 'alpine')
     elif 'NS-BSD' in platform.system():
-        release = re.sub('\-.*\Z', '', ustr(platform.release()))
+        release = re.sub(r'\-.*\Z', '', ustr(platform.release()))
         osinfo = ['nsbsd', release, '', 'nsbsd']
     else:
         try:
-            # dist() removed in Python 3.7
+            # dist() is deprecated in Python 3.5 and docs say will be removed in Python 3.8
             osinfo = list(platform.dist()) + ['']
-        except:
-            osinfo = ['UNKNOWN', 'FFFF', '', '']
+        except Exception:
```

Comment:
is it possible to use a more specific exception?

Reply:
The list of exceptions platform.dist() can raise is not documented. That said, we could make this path for NameError and just let any other exception go to town... this change was due to flake8 identifying "except:" clauses without an exception - so adding "Exception" to it makes it the same thing.